app/actions/client.py

```diff
@@ -213,8 +217,20 @@ async def get_station_conditions(integration, base_url, config, auth):
                         message=parsed_response["message"],
                         status_code=parsed_response["code"]
                     )
-                obs = ConditionsResponse.parse_obj(parsed_response)
-                return obs
+                try:
+                    obs = ConditionsResponse.parse_obj(parsed_response)
+                    return obs
+                except pydantic.ValidationError as e:
+                    msg = f"Response: {parsed_response['conditions']}. Exception: {e}"
+                    logger.error(msg)
+                    await log_action_activity(
+                        integration_id=integration.id,
+                        action_id="pull_station_conditions",
+                        level=LogLevel.WARNING,
+                        title=f"Get station conditions error for station '{config.station.Station_ID}'",
```

Comment:
Maybe change the wording a bit, particularly if you are able to parse the fault payload. Something like this:

```suggestion
                        title=f"Station '{config.station.Station_ID}' reported an error. Fault status: {fault_status} ({fault_description})",
```